.circleci/config.yml

```diff
@@ -97,12 +99,60 @@ jobs:
             coverage run --source=super_gradients -m unittest tests/deci_core_unit_test_suite_runner.py
             coverage report
             coverage html  # open htmlcov/index.html in a browser
-
       - store_artifacts:
           path: htmlcov
 
       - store_artifacts:
           path: ~/sg_logs
+      
+  build_and_publish_beta:
+    description : "building the temp image and pushing to dev ecr"
+    parameters:
+       repo_name:
+         type: string
+         default: deci/super-gradients
+       version:
+          type: string
+          default: ${CIRCLE_BRANCH}
+    docker:
+       - image: cimg/base:stable-20.04
+    resource_class: small
+    steps:
+     - checkout
+     - setup_remote_docker:
+          version: 20.10.7
+          docker_layer_caching: true
+     - deci-common/container_image_build:
+           dockerfile: scripts/Dockerfile
+           repository_name: << parameters.repo_name >>
+           image_tag: "${CIRCLE_BRANCH}"
+           build_args: 'VERSION=<< parameters.version >>'
+           extra_flags: ' --compress'
+     - deci-common/push_docker_image_aws_dev:
+           repository_name: << parameters.repo_name >>
+           image_tag: "${CIRCLE_BRANCH}"
+           additional_tags: "latest"
+
+
+
+  integration_test:
+    description: "running integration test on the code"
+    parameters:
+        version:
+          type: string
+          default: latest
+        image_repo:
+          type: string
+          default: deciai/super-gradients
+    machine: true 
+    resource_class: deci-ai/research-standard
```

Comment:
I think you can run on the container itself, see https://circleci.com/docs/using-docker/

in any case - no need to use the self hosted runners at the moment. I think @shaydeci  said that we did not design the tests for the docker image yet, so we don't know yet if it is necessary

Reply:
@ranrubin true.

Reply:
@strelok899  my comment still stands